_posts/2016-07-20-Automation-testing-with-postman.md

```diff
@@ -0,0 +1,164 @@
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_headers.png)
+<br/>
+<br/>
+Postman also has a few dynamic variables which you can use. For example `{% raw %}{{$guid}}{% endraw %}` a v4 style guid, `{% raw %}{{$timestamp}}{% endraw %}` the current timestamp, `{% raw %}{{$randomInt}}{% endraw %}` a random integer between 0 and 1000. More of those will be added in future releases. But for now let us just simply enter `"x=log(2)"` as the raw content of our request.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_body.png)
+<br/>
+<br/>
+Finally let's hit the "Send" button and if everything goes as expected, we should receive the following response.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_send.jpg)
+<br/>
+<br/>
+Next we are going to write our test, but first let us save our request into a collection. By clicking on the create collection button on the collections tab, the following popup will be displayed.  Simply enter "Calculator" as the name of the collection and hit the create button.
+<br/>
+<br/>
+![Alt text]({{ '/img/postman-2016/create_collection.jpg'| prepend:site.baseurl }})
+<br/>
+<br/>
+Now hit the "Save" button next to the request field. Enter "Log" as the name of the request and select "Calculator" from the dropdown menu.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_save.jpg)
+
+### Writing a test
+A Postman test is essentially JavaScript code which sets values for the special 'tests' object. To know which other objects and libraries are available while writing your test cases, make sure you check the following [link](https://www.getpostman.com/docs/sandbox){:target="_blank"}. Let's copy following code snippet in the Tests sandbox.
+<br/>
+{% highlight bash %}tests["Status code is 200"] = responseCode.code === 200;
+var jsonData = JSON.parse(responseBody);
+tests["Verify result"] = jsonData.result.x === "0.69314718056";{% endhighlight %}
+<br/>
+![Alt text](/img/postman-2016/test.jpg)
+<br/>
+<br/>
+The test will run each time you hit the "Send" button. If you need for example an other functions for setting variables you can write them for example in the pre-request sandbox as shown below:
+<br/>
+<br/>
+![Alt text](/img/postman-2016/custom_function.jpg)
+<br/>
+<br/>
+Here we are using the ‘postman’ object and are calling the setEnvironmentVariable function on it, this allows us to assign the result of our function to a variable on the environment scope for later use.
+
+### Collection Runner
+Let's assume we want to run several tests at once. Postman has a Collection Runner utility that allows us to do that and even thousands of times if we want. To access the runner click on "Runner" in the top header then select  "Calculator" as the collection and "Production" as the environment. We want the runner to do that 2 times so enter 2 in the iteration inputfield like shown in the screenshot below.
```

Comment:
that allows us to do that and even thousands of times if we want. => that allows us to just do that, even thousands of tests if we want. 
(I do not see here what times has to do with: we want to run several tests at once)